store/src/server/api.rs

```diff
@@ -1,27 +1,66 @@
 use crate::config::StoreConfig;
 use crate::db::Db;
 use anyhow::Result;
-use miden_crypto::{merkle::TieredSmt, Felt, FieldElement};
-use miden_node_proto::store;
-use std::net::ToSocketAddrs;
-use tokio::time::Instant;
-use tonic::{transport::Server, Response, Status, Streaming};
+use miden_crypto::{hash::rpo::RpoDigest, merkle::TieredSmt, Felt, FieldElement};
+use miden_node_proto::digest::Digest;
+use miden_node_proto::{
+    store::{self, CheckNullifiersResponse},
+    tsmt::{self, Leaf},
+};
+use std::{net::ToSocketAddrs, sync::Arc};
+use tokio::{sync::RwLock, time::Instant};
+use tonic::{transport::Server, Response, Status};
 use tracing::{info, instrument};
 
 pub struct StoreApi {
     db: Db,
-    nullifier_tree: TieredSmt,
+    nullifier_tree: Arc<RwLock<TieredSmt>>,
 }
 
 #[tonic::async_trait]
 impl store::api_server::Api for StoreApi {
-    type CheckNullifiersStream = Streaming<store::CheckNullifiersResponse>;
-
     async fn check_nullifiers(
         &self,
-        _request: tonic::Request<Streaming<store::CheckNullifiersRequest>>,
-    ) -> Result<Response<Self::CheckNullifiersStream>, Status> {
-        todo!()
+        request: tonic::Request<store::CheckNullifiersRequest>,
+    ) -> Result<Response<CheckNullifiersResponse>, Status> {
+        // Validate the nullifiers and convert them to RpoDigest values. Stop on first error.
+        let nullifiers = request
+            .into_inner()
+            .nullifiers
+            .into_iter()
+            .map(|v| {
+                v.try_into()
+                    .or(Err(Status::invalid_argument("Digest field is not in the modulos range")))
+            })
+            .collect::<Result<Vec<RpoDigest>, Status>>()?;
```

Comment:
Probably not for this PR, but rather thoughts on general approach: since the request is coming from an internal component - do we still need do validation? In this case, the overhead is probably minimal and so it is probably better to do this extra checks anyway. But in case validation logic is more complex, we may want to skip it for internal communication.